src/dataset.cpp

```diff
@@ -574,8 +575,183 @@ Dataset concatenate(const Dataset &d1, const Dataset &d2, const Dim dim) {
   return out;
 }
 
+namespace neutron {
```

Comment:
Just a quick comment, what do you think about moving unit conversions to a seperate file, instead of the main `dataset.cpp` implementation? We might end up having quite a few comversions in the end, it feels like it's kind of polluting `dataset.cpp` a little...